document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java

```diff
@@ -437,6 +442,120 @@ public void testSearch() throws IOException {
+        documents.add(results.next());
+      }
+      Assertions.assertEquals(2, documents.size());
+    }
+
+    // Nested Field exists in the document
+    {
+      Query query = new Query();
+      query.setFilter(new Filter(Op.EXISTS, "attributes.trace_id.value.testKeyExistNested", null));
+      Iterator<Document> results = collection.search(query);
+      List<Document> documents = new ArrayList<>();
+      while (results.hasNext()) {
+        documents.add(results.next());
+      }
+      Assertions.assertEquals(2, documents.size());
+    }
+
+    // Field Not Exists in the document
+    {
+      Query query = new Query();
+      query.setFilter(new Filter(Op.NOT_EXISTS, "attributes.trace_id.value.testKeyExistNested", null));
+      Iterator<Document> results = collection.search(query);
+      List<Document> documents = new ArrayList<>();
+      while (results.hasNext()) {
+        documents.add(results.next());
+      }
+      Assertions.assertEquals(1, documents.size());
+    }
+
+  }
+
+  @Test
+  public void testExistsFilter() throws IOException {
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+    collection.upsert(new SingleValueKey("default", "testKey1"),
+            Utils.createDocument(
+                    ImmutablePair.of("id", "testKey1"),
+                    ImmutablePair.of("name", "abc1"),
+                    ImmutablePair.of("size", -10.2),
+                    ImmutablePair.of("isCostly", false)));
+    collection.upsert(new SingleValueKey("default", "testKey2"),
+            Utils.createDocument(
+                    ImmutablePair.of("id", "testKey2"),
+                    ImmutablePair.of("name", "abc2"),
+                    ImmutablePair.of("size", 10.4),
+                    ImmutablePair.of("isCostly", false)));
+    collection.upsert(new SingleValueKey("default", "testKey3"),
+            Utils.createDocument(
+                    ImmutablePair.of("id", "testKey3"),
+                    ImmutablePair.of("name", "abc3"),
+                    ImmutablePair.of("size", 30),
+                    ImmutablePair.of("isCostly", false),
+                    ImmutablePair.of("city", "bangalore")));
+    collection.upsert(new SingleValueKey("default", "testKey4"),
+            Utils.createDocument(
+                    ImmutablePair.of("id", "testKey4"),
+                    ImmutablePair.of("name", "abc4"),
+                    ImmutablePair.of("size", 30),
+                    ImmutablePair.of("isCostly", false),
+                    ImmutablePair.of("city", null)));
+    Query query = new Query();
+    query.setFilter(new Filter(Op.EXISTS, "city", true));
+    Iterator<Document> results = collection.search(query);
+    List<Document> documents = new ArrayList<>();
+    while (results.hasNext()) {
+      documents.add(results.next());
+    }
+    Assertions.assertEquals(documents.size(), 2);
+  }
+
+  @Test
+  public void testNotExistsFilter() throws IOException {
```

Comment:
Can you add the same test in MongoDocStoreTest?

Reply:
Added